packages/manager/apps/web/client/app/dns-zone/zone.routing.js

```diff
@@ -23,6 +21,52 @@ export default /* @ngInject */ ($stateProvider) => {
       defaultFilterColumn: () => 'name',
       header: /* @ngInject */ ($translate) => $translate.instant('zones_title'),
       customizableColumns: () => true,
+      columns: /* @ngInject */ ($translate) => [
+        {
+          property: 'name',
+          title: $translate.instant('zones_domain_name'),
+          template: `<a data-ng-href="{{ $ctrl.getServiceNameLink($row) }}" data-ng-bind="$row.name"></a>`,
```

Comment:
**nitpick:** `resolve` is executed before the controller is initialised, so using a function defined in the controller can be misleading. Can you define `getServiceNameLink` in the resolve and inject it here and use.

Reply:
Hello @anooparveti 

Thank you for the review and the suggestion regarding the `getServiceNameLink` function. 
I understand the intention behind moving this function to the resolve section to ensure all dependencies are ready before the controller initializes. 
However, there are a couple of reasons why this approach might not be feasible in this context:

1. Execution Context: The getServiceNameLink function relies on the `$state` service to dynamically generate state-based links, which inherently requires access to the current execution context, including the controller's scope and potentially the current application state. Since the resolve block is executed before the controller is instantiated, it doesn't have access to the controller's context or any runtime state data that might influence link generation.

2. Service Usage: While we could inject the `$state` service into the resolve block, using it to dynamically generate links within resolve is somewhat unconventional. The primary purpose of resolve is to prepare and provide data or static dependencies necessary for the controller, rather than to serve dynamic functions like link generation based on the application's runtime state.

Given these considerations, `getServiceNameLink` might be more appropriately defined within the controller or as a method of a service that can be injected into the controller. 
This would ensure that it has access to the necessary context and services to function as intended.

I hope this clarifies why moving `getServiceNameLink` to the resolve section might not be the best approach in this scenario. If there are other concerns or alternative suggestions, I'm open to discussing them further.

Reply:
Hello @anooparveti ,

Thank you for the follow-up and the suggestion to directly inject `getServiceNameLink` into the columns definition. 
I tried implementing the suggested change.

However, I encountered an issue with this approach. It seems that the function `getServiceNameLink` isn't directly accessible within the template string in this context. This might be due to the way AngularJS compiles and links templates, where the scope within `ng-href` might not have direct access to resolved dependencies in the same way the controller does.

Given this, while the function is indeed defined in the `resolve` section, it appears that its direct use within the template string doesn't work as expected. This could be related to AngularJS's scope hierarchy and how it resolves bindings in templates, especially when it comes to functions and their contexts.

To maintain the functionality and ensure that the links are correctly generated, it might be necessary to keep the `$ctrl.getServiceNameLink` reference in the template. This ensures the function is called within the correct scope and has access to all necessary dependencies.

If there's a workaround or a different approach that could make this work as intended, I'd be keen to try it out.

Thank you again for your review and guidance.